docs(chart): Add KB about how to bind chart to DataTable (#1246)
telerik/blazor-docs: knowledge-base/chart-datatable-binding.md

---

> title: How to use DataTable as source of data for the Chart

Reviewer: This title is long and uses keywords, which the customer didn't, so they may be unnecessary at least in the title ("use", "source"). At the same time, the customer used the keyword "bind", which we mention just once in the whole article in another grammar form ("binding"). The same applies for the word "convert" in the meta description - the customer is converting apps, not data sources.

Overall, our KBs must capture the customer context, which means to use the keywords used by customers. We can add more keywords, but not at the cost of the required ones.

description: How to convert DataTable to List and use it as data source for the Chart
type: how-to
page_title: How to use DataTable as data source for the Chart

> slug: chart-datatable-binding

Reviewer: What is more likely for developers to search - "chart bind (to) datatable" or "chart datatable binding". Think about it and if needed, adjust the slug and the file name. I am specifically interested in the grammar form of "bind..."

position: 

> tags: 

Reviewer: ```suggestion
tags: telerik, blazor, chart, datatable
```

ticketid: 1589460 
res_type: kb
---

## Environment
<table>
    <tbody>
	    <tr>
	    	<td>Product</td>
	    	<td>Charts for Blazor</td>
	    </tr>
    </tbody>
</table>

## Description

> I want to use DataTable as a data source for the Chart series.

Reviewer: It is acceptable to say the same thing 1-2 times, so that you include more / all desired keywords.
```suggestion
How to data bind the Telerik Blazor Chart to a DataTable?

How to use a DataTable as a data source for the Chart?
```

## Solution

The Telerik UI for Blazor Chart does not support DataTable binding out of the box. To use DataTable instance as a data source for the series of the Chart you have to convert the DataTable to IEnumerable<T> e.g. List<T>.

> Here is a simple example:

Reviewer: No need for sentences, which do not add value. Inst
```suggestion
>caption Bind Chart to a converted DataTable
```

````CSHTML
@using System.Data;
@using System.Dynamic;

<TelerikChart>
    <ChartSeriesItems>
        <ChartSeries Type="ChartSeriesType.Pie"

>                      Data="Data"

Reviewer: We use @ in such places. See our docs and demos code style guide in the `blazor` repo wiki.

```suggestion
                     Data="@Data"
```

                     Field="Value"
                     CategoryField="Name">
            <ChartSeriesLabels Visible="true"></ChartSeriesLabels>
        </ChartSeries>
    </ChartSeriesItems>
</TelerikChart>

@code {
    private List<ExpandoObject> Data { get; set; } = ConvertDataTable(GetData(), "Value", "Name");

    private static DataTable GetData()
    {
        var table = new DataTable();
        table.Columns.Add("Value", typeof(double));
        table.Columns.Add("Name", typeof(string));
        table.Rows.Add(1, "John");
        table.Rows.Add(5, "Peter");
        table.Rows.Add(6, "Paul");
        return table;
    }

    private static List<ExpandoObject> ConvertDataTable(DataTable dt, string field, string categoryField)
    {
        List<ExpandoObject> data = new List<ExpandoObject>();
        foreach (DataRow row in dt.Rows)
        {
            var expando = new ExpandoObject() as IDictionary<string, object>;
            expando.Add(field, row[field]); 
            expando.Add(categoryField, row[categoryField]);
            data.Add((ExpandoObject)expando);
        }
        return data;
    }
}
````

> 

Reviewer: Consider a "See Also" section with other useful articles or demos related to Chart binding. Optionally, consider linking to this article from a Chart docs article, but only if you find a suitable article and place.
